src/PackagistClient.php

```diff
@@ -84,6 +85,94 @@ public function getStatistics(): ?array
         return $this->request('statistics.json');
     }
 
+    /**
+     * Get security vulnerability advisories for specific packages and/or which have been updated since some timestamp.
+     *
+     * If $filterByVersion is true, the $packages array must have package names as keys and versions as values.
+     * If it is false, the $packages array can contain package names as values.
+     *
+     * @throws InvalidArgumentException if no packages and no updatedSince timestamp are passed in
+     */
+    public function getAdvisories(array $packages = [], ?int $updatedSince = null, bool $filterByVersion = false): array
+    {
+        if (count($packages) === 0 && $updatedSince === null) {
```

Comment:
The readability of this code can be improved.

Please avoid using `&&` and split this into multiple smaller functions so the comments can be removed.

Reply:
Inline comments removed.
That specific `&&` is not longer required. I've simply removed the ability to get all packages by udpatedsince because this PR is becoming burdonsome to develop. If someone wants that functionality they can add it separately afterward.